### DISCUSSION TITLE
Как назначить доступ к циклу через #creator_id, если user_ов больше 20 человек???

### DISCUSSION THREAD

**Discussion by IrinaG1090** [totumonline/totum-ru-issues-and-discussions#479] [Как это сделать?] [answered]
https://github.com/totumonline/totum-ru-issues-and-discussions/assets/23610956/ff8c06f6-aa0a-480d-a955-fa67aee74244

Есть задача(цикл). Создана одним оператором (всего операторов 12 человек). Оператор видит только свой цикл. Эту задачу (цикл) руководитель назначает менеджеру(менеджеров 5-10 человек) и статус у менеджера меняется, видно на видео. Теперь этот цикл видят: менеджер назначенный и оператор-автор.Код написан для одного оператора, у которого id=ХХ. Если попробовать прописать возможность для всех операторов и менеджеров, я использовала сравнение выражений 
(например, 10 < #creator_id < 40) - не сработало. Пробовала конструкцию COND - тоже не работает. 

Как отладить код ? как будет выглядеть в корректности? Как прописать период выборки операторов по #creator_id ? Как их перечислить или считать?

прикрепляю видео-пример. 

На нем пробую назначить цикл от одного оператора - конкретному менеджеру. Также показываю что мои догадки не работают, если делать сравнение (например, 10 < #creator_id < 40) Как сделать чтоб любой из операторов мог назначить задачу менеджеру? Надеюсь, понятно обьяснила .....

**Comment by ttm-alexeypolunin** [comment]
Ну я не буду здесь в этом разбираться  — приходите на консультацию, посмотрим.

› видео не смотрел

› пока не ставлю answered, может кто ответит еще

**Reply by ttm-alexeypolunin** [reply]
Из советов:

Для открытия циклов на других людей используется Тип доступа к циклам (префильтр нет тк он не блокирует доступ внутрь цикла) https://ru.docs.totum.online/roles-and-users#cycle-type-access

Видео: https://ru.docs.totum.online/how-to-configure-loop-accesses-for-users

Далее доступы передаются через таблицу users, но в вашем случае наверное добавлением в creator_id.

Префильтры сейчас работают только по полному соответствию. Перейдем на коммерческие лицензии, может и доделаем расширение префильтров — оно уже как 5 лет запланировано.

**Reply by IrinaG1090** [reply]
Благодарю за советы, но эти настройки сделаны были вначале создания идеи циклов)))

возможно я невнятно объяснила. 

При назначении менеджеру задачу в creator_id срабатывает изменение статуса в другой ячейке, если автор цикла с id = 10, а если автор цикла с id = 11 или 12 или 13 или 14, то это это изменение статуса не срабатывает, потому что для этих других id ничего не прописано. Я пыталась прописать в creator_id сразу все нужные id, которые есть в users. Но пишет ошибку. 

Я уверена решение есть очевидное, просто у меня видимо глаз замылился

**Reply by IrinaG1090** [reply]
То есть я писала вот так :
if(condition: #creator_id = 10; condition: #status_obrabotki = 2; then: $s1; else: $f1
s1 = set(table ‘xxx’ field: ‘menedger’ = 4)


Тут я охватываю только один id, который равен 10. 

А как охватить несколько id ? Например, количество id от 10 до 20? id = 11, id = 12, id = 13 и так далее. 

Пробовала через COND - не получилось, пробовала 10 < #creator_id < 20   - тоже не вышло, пробовала каждый вручную прописать - тоже не вышло. 


Может я чего не вижу …..

**Reply by erpic-ru** [reply, marked as answer]
Разве так не работает условие?:
```
if(condition: #creator_id >= 10; condition: #creator_id <= 20;  condition: #status_obrabotki = 2; then: $s1; else: $f1
s1 = set(table ‘xxx’ field: ‘menedger’ = 4)
```

Но вообще лучше на роли завязываться, чем на id пользователей

**Reply by ttm-alexeypolunin** [reply]
Это пример реального синтаксиса? Так не работает

```
s1 = set(table ‘xxx’ field: ‘menedger’ = 4)
```

работает так:
```
s1: set(table ‘xxx’; field: ‘menedger’ = 4)
```

так тоже

```
condition: id = 11, id = 12, id = 13
```
Синтаксис работает так:

```
condition: #id = 11; condition:  #id = 12; condition: #id = 13;
```
или

```
condition: #id = json`[11,12,13]`;
```

так тоже не работает (эрпик написал выше пример)

```
10 < #creator_id < 40
```

Надо помнить, что Селект хранит строки и они сравниваются больше-меньше как строки, те "9" > "20"

```
condition: #creator_id >= 10; condition: #creator_id <= 20;
```

Если вам выпадает ошибка, посмотрите, что в ней написано.

Также есть отладка, смотрите что происходит при выполнении действия https://ru.docs.totum.online/debugging#sx-enabling-debugging

**Reply by IrinaG1090** [reply]
благодарю за советы, попробую применить

**Comment by IrinaG1090** [comment]
Всё заработало!!!! круто помогли, советы топ, пушка-бомба-ракета!!!!! благодарю :1st_place_medal:
